plugins/AuthUI/src/commonMain/kotlin/io/github/temk0/supabase/authui/AuthUI.kt

```diff
@@ -0,0 +1,56 @@
+package io.github.temk0.supabase.authui
+
+import io.github.jan.supabase.SupabaseClient
+import io.github.jan.supabase.gotrue.gotrue
+import io.github.jan.supabase.gotrue.providers.Google
+import io.github.jan.supabase.gotrue.providers.builtin.IDToken
+import io.github.jan.supabase.plugins.SupabasePlugin
+import io.github.jan.supabase.plugins.SupabasePluginProvider
+
+
+sealed interface AuthUI : SupabasePlugin {
+
+    data class Config(
+        var loginConfig: LoginConfig? = null
+    ) : SupabasePlugin
+
+    companion object : SupabasePluginProvider<Config, AuthUI> {
+
+        override val key: String = "authui"
+
+        const val API_VERSION = 1
+
+        override fun create(supabaseClient: SupabaseClient, config: Config): AuthUI {
+            return AuthUIImpl(config, supabaseClient)
+        }
+
+        override fun createConfig(init: Config.() -> Unit): Config {
+            return Config().apply(init)
+        }
+    }
+
+    suspend fun loginWithGoogle(idToken: String)
```

Comment:
```suggestion
    suspend fun loginWithGoogle(idToken: String, nonce: String)
```
I suggest that this function is only in the implementation and not in the interface, as this should only be used internally here.

Reply:
And for the nonce, maybe put a `useNonce` property in the config and then automatically apply a nonce to the onetap client & login with idtoken